src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingDocumentControllerTest.java

```diff
@@ -85,12 +93,47 @@ public void generateHearingDocumentFormC() throws Exception {
                 .andExpect(jsonPath("$.data.formC.document_binary_url", is(BINARY_URL)));
     }
 
+    @Test
+    public void shouldSendSolicitorEmailWhenAgreed() throws Exception {
+        mvc.perform(post(HEARING_DOCUMENT_CALLBACK_URL)
+            .content("{\n"
+                + "  \"case_details\": {\n"
+                + "    \"case_data\": {\n"
+                + "      \"solicitorAgreeToReceiveEmails\": \"Yes\"\n"
```

Comment:
this could be a method, like: `prepareBodyWithSolicitorAgreedReceiveEmails(String value)`

```
mvc.perform(post(HEARING_DOCUMENT_CALLBACK_URL)
            .content(prepareBodyWithSolicitorAgreedReceiveEmails("yes")
```

or:

```
mvc.perform(post(HEARING_DOCUMENT_CALLBACK_URL)
            .content(prepareBodyWithSolicitorAgreedReceiveEmails("no")
```
in next test

Reply:
Yes it could be. I initially thought to do it like this but decided against it as i had heard previous complaints of people adding too many private functions that were not needed. However im happy to add it in